lib/services/presets.go

```diff
@@ -29,6 +29,68 @@ import (
 	"github.com/gravitational/teleport/lib/modules"
 )
 
+func NewPresetAutomaticAccessApproverRole() types.Role {
+	enterprise := modules.GetModules().BuildType() == modules.BuildEnterprise
+	if !enterprise {
+		return nil
+	}
+	role := &types.RoleV6{
+		Kind:    types.KindRole,
+		Version: types.V7,
+		Metadata: types.Metadata{
+			Name:        teleport.PresetAutomaticAccessApprovalRoleName,
+			Namespace:   apidefaults.Namespace,
+			Description: "Approves any access request",
+			Labels: map[string]string{
+				types.TeleportSystemResourceLabel: types.IsSystemResource,
+				types.TeleportResourceRevision:    "1",
+			},
+		},
+		Spec: types.RoleSpecV6{
+			Allow: types.RoleConditions{
+				ReviewRequests: &types.AccessReviewConditions{
+					Roles: []string{"*"},
+				},
+			},
+		},
+	}
+	return role
+}
+
+// NewPresetAutomaticAccessBotUser returns a new User that has (via the
+// the `PresetAutomaticAccessApprovalRoleName` role) the right to automatically
+// approve any access requests.
+//
+// This user must not:
+//   - Be allowed to log into the cluster
+//   - Show up in user lists in any UI (including Web, `tsh`, `tctl`, etc)
```

Comment:
Couple notes here:

- I think this user should not be able to log into anything as it is? If it only has the "reviewer" role.
- IMO it should be ok for this to show up in the user listings. Otherwise it might be difficult to troubleshoot, debug, etc.

Reply:
Fair enough. I wonder if there's something we can use the labels to colour code them in the UI or something.